haystack/document_stores/milvus1.py

```diff
@@ -105,6 +108,13 @@ def __init__(
                                     exists.
         :param isolation_level: see SQLAlchemy's `isolation_level` parameter for `create_engine()` (https://docs.sqlalchemy.org/en/14/core/engines.html#sqlalchemy.create_engine.params.isolation_level)
         """
+        deprecation_message = (
+            "Milvus1DocumentStore is deprecated and will be removed in a future version. "
+            "Please consider switching to Milvus2 or to another DocumentStore."
+        )
+        warnings.simplefilter("always", DeprecationWarning)
```

Comment:
I saw that our other DeprecationWarnings don't set this filter. They just use:
```
warnings.warn(
                "The 'vector_dim' parameter is deprecated, " "use 'embedding_dim' instead.", DeprecationWarning, 2
            )
```
Downside(?) of this filter setting is that users can't control whether they see this warning or not - it will always be shown. I would say it's an advantage if users can turn it off. However, it's better to show it always than never and I guess without this line of code you didn't get the warning when you tested it?

Reply:
According to [warnings documentation](https://docs.python.org/3/library/exceptions.html#DeprecationWarning), DepreactionWarnings are ignored by default, so without this line, the user would not see the warning unless they change the bahavior of `warning` themself. I agree that this is not optimal - as you mentioned, the user can't control whether this warning is shown or not

Would you suggest removing the `warnings.simplefilter("always", DeprecationWarning)` statement?

Reply:
Let's see what our options are here. I am thinking that this line of code can result in many warnings shown to the user not only from Haystack but also its dependencies. sklearn caused an issue like this here: https://github.com/scikit-learn/scikit-learn/issues/8182
Following up on this, maybe you could check https://docs.python.org/3/library/exceptions.html#FutureWarning as an alternative?

Reply:
Great learning! Difference of [DeprecationWarning](https://docs.python.org/3/library/exceptions.html#DeprecationWarning) and [FutureWarning](https://docs.python.org/3/library/exceptions.html#FutureWarning) was also new to me. This PR is ready to be merged from my side. 👍